lib/Parser/ptree.h

```diff
@@ -309,6 +311,19 @@ class ParseNodeInt : public ParseNode
     DISABLE_SELF_CAST(ParseNodeInt);
 };
 
+// bigint constant
+class ParseNodeBigInt : public ParseNode
+{
+public:
+    ParseNodeBigInt(charcount_t ichMin, charcount_t ichLim, IdentPtr pid);
+
+    IdentPtr const pid;
```

Comment:
Not saying you should change this but I wonder if, from a testability standpoint, it makes sense for a stand-alone BigInt implementation that we can store here instead of an identptr, and have JavascriptBigInt be the JS wrapper around this implementation. It makes it more in line with ParseNodeInt/ParseNodeFloat but I guess that approach also has other limitations (namely, what allocator to allocate the uint32 array if it's allocated at parse time and needs to be alive for the duration of the JavascriptBigInt). What was your thought process to factor it in this way?

Reply:
I was debating about this. Even though BigInt is a numeric primitive, in the parse tree, I would view it more of String-ish node. That's why I chosed to follow the pattern of ParseNodeStr. Here is some advantages that I can think of.
- take advantage of hashing string in the constant table, I don't know how can we hash BigInt though.
- we might able to defer the convert string to array (BigInt underline) until we need to do some calculations.

Also, it is nice if it is in line with ParseNodeInt/ParseNodeFloat but I don't see why we need to carry value of BigInt in the parse node tree. We only care about the value until we do some calculations. (debatably, we need to maintain one more key for string in the hash table). 

Overall, it is not too much code change to refactor it in your suggestions. We can do it later if needed.